src/LtiServiceConnector.php

```diff
@@ -75,38 +76,60 @@ public function getAccessToken(array $scopes)
         return $tokenData['access_token'];
     }
 
-    public function makeServiceRequest(array $scopes, string $method, string $url, string $body = null, $contentType = 'application/json', $accept = 'application/json')
-    {
+    public function makeServiceRequest(
+        array $scopes,
+        string $method,
+        string $url,
+        string $body = null,
+        $contentType = 'application/json',
+        $accept = 'application/json',
+        bool $hasRetried = false
```

Comment:
```suggestion
        bool $shouldRetry = true
```
And of course, switch the one down below to `false` when it calls the method a second time.

I find it easier to read this way, but that's just my opinion. If you don't like it, feel free to resolve this comment.